server/datastore/mysql/hosts.go

```diff
@@ -1139,6 +1139,30 @@ func (ds *Datastore) DeleteHosts(ctx context.Context, ids []uint) error {
 	return nil
 }
 
+func (ds *Datastore) FailingPoliciesCount(ctx context.Context, host *fleet.Host) (uint, error) {
+	if host.FleetPlatform() == "" {
+		// We log to help troubleshooting in case this happens.
+		level.Error(ds.logger).Log("err", fmt.Sprintf("host %d with empty platform", host.ID))
+	}
+
+	query := `
+		SELECT SUM(1 - pm.passes) AS n_failed
+		FROM policies p INNER JOIN policy_membership pm ON (p.id = pm.policy_id AND pm.host_id = ?)
+		WHERE (p.team_id IS NULL OR p.team_id = (select team_id from hosts WHERE id = ?))
```

Comment:
I'm a little confused why this is needed. Can a host be in policy_membership if the associated policy has a team id and the host is not part of that team? What is being missed when we do 

```
INNER JOIN policy_membership pm ON (p.id = pm.policy_id AND pm.host_id = ?)
```

Reply:
Moved convo here https://github.com/fleetdm/fleet/pull/7530/files#r962932254 - as this PR is only for changes related to Fleet Desktop.